agent/app/agent.go

```diff
@@ -371,6 +378,22 @@ func (agent *ecsAgent) newTaskEngine(containerChangeEventStream *eventstream.Eve
 	return previousTaskEngine, currentEC2InstanceID, nil
 }
 
+func (agent *ecsAgent) initMetricsEngine() string {
+	// In case of a panic during set-up, we will recover quiently and resume
+	// normal Agent execution.
+	defer func() {
+		if r := recover(); r != nil {
+			seelog.Errorf("MetricsEngine Set-up panicked. Recovering quietly: %s", r)
+		}
+	}()
+
+	// We init the global MetricsEngine before we publish metrics
+	metrics.MustInit(agent.cfg)
+	prometheusContainerID := monitor.InitPrometheusContainer(agent.ctx, agent.cfg, agent.dockerClient)
+	metrics.PublishMetrics()
```

Comment:
do you need to skip this when the above InitPrometheusContainer fails?

Reply:
PublishMetrics execution is not linked to InitPrometheusContainer. If Agent cannot start the container, customers may still want the endpoint exposed from the Agent side. Customers can chose to restart Agent to bring our Prometheus container back up.